test/test_distributions.py

```diff
@@ -982,3 +982,29 @@ def test_enumerate_support_smoke(jax_dist, params, support, batch_shape, expand)
     if expand:
         expected = np.broadcast_to(expected, support.shape + batch_shape)
     assert_allclose(actual, expected)
+
+
+@pytest.mark.parametrize('jax_dist, sp_dist, params', CONTINUOUS + DISCRETE)
+@pytest.mark.parametrize('prepend_shape', [
+    (),
+    (2, 3),
+])
+@pytest.mark.parametrize('sample_shape', [
+    (),
+    (4,),
+])
```

Comment:
I think we should also add tests to check the interaction of `.to_event` and `.expand` and `log_prob` (you could take a look at `check_sample_shapes` in `test_distributions` in Pyro which is used to test many of these corner cases), but feel free to do that in a separate PR since you are currently blocked by this. Also, scipy distributions don't consistently follow the tensor shaping semantics that we have in pytorch, so it might be best to not take that as a reference.

Reply:
Thanks, @neerajprad ! I'll add it to Masked PR.